### DISCUSSION TITLE
Inertial Oscillations Simulation

### DISCUSSION THREAD

**Discussion by claudioiturra** [CliMA/Oceananigans.jl#3635] [Computational science!]
Hi, I'm new to Oceananigans trying to simulates inertial oscillations (wind induced-inertial oscillation) to evaluate propagation depending on stratification. I will extremely appreciate any advice on how to start with this journey, I will appreciate any comments. Thanks to all

**Comment by glwagner** [comment]
Here's some code to simulate a _pure_ inertial wave (this is often called an "inertial oscillation", but "wave" is definitely a better term):

```julia
using Oceananigans
using Oceananigans.Units
using GLMakie

Nz = 100
Lz = 400
f = 1e-4

grid = RectilinearGrid(size=Nz, z=(-Lz, 0), topology=(Flat, Flat, Bounded))
coriolis = FPlane(; f)
model = HydrostaticFreeSurfaceModel(; grid, coriolis, tracers=(), buoyancy=nothing)

h = 50 # vertical scale of the inertial wave
uᵢ(z) = exp(-z^2 / 2h^2)
set!(model, u=uᵢ)

T = 2π / f # inertial period
Δt = 1e-2 * T
simulation = Simulation(model; Δt, stop_time=4T)

data = []

function save_data!(sim)
    u, v, w = sim.model.velocities
    ui = interior(u, 1, 1, :)
    vi = interior(v, 1, 1, :)
    push!(data, (t=time(sim), u=deepcopy(ui), v=deepcopy(vi)))
    return nothing
end

add_callback!(simulation, save_data!, TimeInterval(0.02T))

run!(simulation)

fig = Figure()
ax = Axis(fig[1, 1], xlabel="Velocities", ylabel="z (m)")

Nt = length(data)

n = Observable(1)
un = @lift data[$n].u
vn = @lift data[$n].v
tn = @lift data[$n].t

z = znodes(model.velocities.u)

lines!(ax, un, z, label="u")
lines!(ax, vn, z, label="v")
xlims!(ax, -1.1, 1.1)
axislegend(ax, position=:rb)

record(fig, "inertial_oscillation.mp4", 1:Nt, framerate=24) do nn
    n[] = nn
end
```

which produces

https://github.com/CliMA/Oceananigans.jl/assets/15271942/40a6cc06-7dda-4475-8823-0701e4c6ee08

To simulate the vertical propagation of a near-inertial wave in a stratified fluid, you'll need to make the simulation 2D, add some horizontal structure (for example `uᵢ(y, z) = cos(k * y) * exp(-z^2 / 2h^2)` where `k` is a wavenumber), and add buoyancy + a stratified initial condition. A good starting point is probably the [internal wave example](https://github.com/CliMA/Oceananigans.jl/blob/main/examples/internal_wave.jl) which has many of these ingredients.

**Reply by glwagner** [reply]
And note that the internal wave in the example can indeed be made near-inertial by increasing `m` and decreasing `k`:

https://github.com/CliMA/Oceananigans.jl/blob/7f2d512ceaa2ce3fec3132cf0b08ab39344270d9/examples/internal_wave.jl#L77-L79

due to the dispersion relation

https://github.com/CliMA/Oceananigans.jl/blob/7f2d512ceaa2ce3fec3132cf0b08ab39344270d9/examples/internal_wave.jl#L83

which limits to `ω² = f^2` as `k / m` goes to 0.

But you may be interested in different initial condition so I recommend preparing your own script.

**Reply by claudioiturra** [reply]
Dear @glwagner Thank you so much for the code and notes. I apologize for not seeing the response on time. For some reason, I couldn't receive the response alert. I've tested the code and it works great. I'm currently understanding all the parameters in the internal wave example, definitely need to go back to the books. Appreciate your time in respond to my question.

**Reply by glwagner** [reply]
No need to apologize. Godspeed in your journey through physics!
